core/common/src/main/java/alluxio/wire/MasterInfo.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Master information.
+ */
+@NotThreadSafe
+public final class MasterInfo {
+  /** Master's address. */
+  private Address mAddress;
+  /** The id of the master. */
+  private long mId;
+  /** Master's last updated time in ms. */
+  private long mLastUpdatedTimeMs;
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   */
+  public MasterInfo() {}
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param id the master id to use
+   * @param address the master address to use
+   * @param lastUpdatedTimeMs the master lastUpdatedTimeMs to use
+   */
+  public MasterInfo(long id, Address address, long lastUpdatedTimeMs) {
+    mAddress = Preconditions.checkNotNull(address, "address");
+    mId = id;
+    mLastUpdatedTimeMs = lastUpdatedTimeMs;
+  }
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param id the master id to use
+   * @param address the master address to use
+   */
+  public MasterInfo(long id, Address address) {
+    mAddress = Preconditions.checkNotNull(address, "address");
+    mId = id;
+    mLastUpdatedTimeMs = System.currentTimeMillis();
+  }
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param masterInfo the masterInfo need to be cloned
+   * @return the the master information
+   */
+  public MasterInfo clone(MasterInfo masterInfo) {
+    mAddress = masterInfo.getAddress();
+    mId = masterInfo.getId();
+    mLastUpdatedTimeMs = masterInfo.getLastUpdatedTimeMs();
+    return this;
+  }
```

Comment:
In the 2nd or 3rd constructors, call `this(a, b, c)` instead of rewriting assignments